src/openssl.c

```diff
@@ -121,6 +135,76 @@ _libssh2_rsa_new(libssh2_rsa_ctx ** rsa,
                  unsigned long e2len,
                  const unsigned char *coeffdata, unsigned long coefflen)
 {
+#ifdef USE_OPENSSL_3
+    int ret = 0;
+    EVP_PKEY_CTX *ctx;
+    OSSL_PARAM params[4];
+    int param_num = 0;
+    unsigned char *nbuf = NULL;
+    unsigned char *ebuf = NULL;
+    unsigned char *dbuf = NULL;
+
+    (void)pdata;
```

Comment:
p, q, e1, e2, and the coefficient can all be calculated as needed by n, e, and/or d, and OpenSSL 3 appears to do this anyway. In my testing, we have not needed to set these paramaters for OpenSSL 3 code.